lib/utils/tls.go

```diff
@@ -54,6 +67,71 @@ type CertificateName struct {
 	OrganizationalUnit []string `json:"org_unit"`
 }
 
+// GenerateSelfSignedCert generates a self signed certificate that
+// is valid for given domain names and ips, returns PEM-encoded bytes with key and cert
+// Generates a certificate that is compatible with the MacOS requirements described at:
+// https://support.apple.com/en-us/HT210176
+func GenerateSelfSignedCert(hostNames []string) (*teleutils.TLSCredentials, error) {
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	notBefore := time.Now()
```

Comment:
IIRC it's generally recommended to set this time slightly in the past, to ensure the cert will be accepted by clients with clock drift errors. I believe letsencrypt backdates by 1 hour.